packages/one-app-hmr/lib/webpack/dll.js

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import webpack from 'webpack';
+import merge from 'webpack-merge';
+
+import {
+  isDevelopment,
+  getContext,
+  getStaticPath,
+  createOneAppExternals,
+  createMinifyConfig,
+} from './utility';
+import {
+  jsxLoader,
+} from './loaders';
+
+// eslint-disable-next-line import/prefer-default-export
+export function createDLLConfig({
+  isDev = isDevelopment(),
+  dllAsReference = false,
+  dllName = 'externals',
+  dllPath = getStaticPath(`.${dllName}.dll.json`),
+  dllVendors = createOneAppExternals(),
+} = {}) {
+  return {
+    ...(dllAsReference
+      ? {}
+      : merge(
+        createMinifyConfig({ isDev }),
+        {
+          mode: isDev ? 'development' : 'production',
+          entry: { [dllName]: dllVendors },
+          output: {
+            path: getStaticPath(),
+            filename: 'vendor/[name].js',
+            library: dllName,
+          },
+          module: {
+            rules: [jsxLoader],
+          },
+        }
+      )),
+    plugins: dllAsReference
+      ? [
+        new webpack.DllReferencePlugin({
+          context: getContext(),
+          name: dllName,
+          manifest: dllPath,
+        }),
+      ]
+      : [
+        new webpack.DllPlugin({
+          context: getContext(),
+          name: dllName,
+          path: dllPath,
+        }),
+      ],
+  };
+}
```

Comment:
The DLL plugin works in two build steps; one for building the DLL with `DllPlugin` and the `DllReferencePlugin` is used in the consuming build. This function abstracts that for us and allows us to use it in both places.